### PR TITLE
docs(block): remove doc/block.txt

### PR DESCRIPTION
**Pull request by SimonRichardson**
The block.txt was out of date regarding what the code should look like and what commands to use to enable and disable the blocks. The doc was written like a specification and we have a new location for that information.

I've dispersed the information into better locations, either in doc.go files or on methods that deal with the blocking actions. This is more likely going to be kept up to date from a developer's perspective.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages

## Links

**Jira card:** [JUJU-4989](https://warthogs.atlassian.net/browse/JUJU-4989)

[JUJU-4989]: https://warthogs.atlassian.net/browse/JUJU-4989?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apiserver/common/block.go [hunk 1]
```diff
@@ -10,14 +10,20 @@ import (
...
 
 // BlockCheckerInterface defines methods of BlockChecker.
 type BlockCheckerInterface interface {
+	// ChangeAllowed checks if change block is in place.
 	ChangeAllowed() error
+	// RemoveAllowed checks if remove block is in place.
 	RemoveAllowed() error
+	// DestroyAllowed checks if destroy block is in place.
 	DestroyAllowed() error
 }
```
Comment (tmihoc) [inline apiserver/common/block.go:29]: How does "in place" connect to "`Allowed`"?

File: cmd/juju/block/protection.go [hunk 2]
```diff
@@ -86,8 +86,13 @@ var blockedMessages = map[Block]string{
...
-// ProcessBlockedError ensures that correct and user-friendly message is
-// displayed to the user based on the block type.
+// ProcessBlockedError handles the error returned by the API server, which
+// may indicate that the operation is blocked.
+// Processing of blocked commands is done by the API server based on the
+// user configuration via `disable-command` command. The user can
+// `enable-command` the operation by enabling the block type.
+// When an operation is blocked the API server returns a blocked error code
+// that returns a user-friendly message to the client.
 func ProcessBlockedError(err error, block Block) error {
```
Comment (tmihoc) [inline cmd/juju/block/protection.go:96]: Where does `disabled-commands` fit in here?

**Review by tmihoc — APPROVED** [review]
This looks much better -- more detailed and more relevant -- than the txt we used to have.

I had a couple of small questions -- feel free to ignore if you think they're actually not an issue.

File: apiserver/common/block.go (hunk 1 above)
Comment (SimonRichardson) [reply apiserver/common/block.go:29]: Figure of speech. Is there a blocking type in persisted to state?

File: cmd/juju/block/protection.go (hunk 2 above)
Comment (SimonRichardson) [reply cmd/juju/block/protection.go:96]: Explains to the observer that the blocked error arrived from user-initiated command.

**Pull request merged**